lib/cudadrv/memory.jl

```diff
@@ -613,6 +613,19 @@ end
 # auxiliary functionality
 #
 
+## memory pinning
+const __pinned_memory = Dict{Ptr, WeakRef}()
+function pin(a::Base.Array, flags=0)
+    # use pointer instead of objectid?
+    ptr = pointer(a)
+    if haskey(__pinned_memory, ptr) && __pinned_memory[ptr].value !== nothing
+        return nothing
+    end
+    ad = Mem.register(Mem.Host, pointer(a), sizeof(a), flags)
+    finalizer(_ -> Mem.unregister(ad), a)
```

Comment:
my old reservations still hold (recycled arrays triggering re-registration), but if it works for you we can start with this.

Reply:
Should absolutely revisit once we can figure out what the deal is with the missing regions